src/cargo/ops/cargo_compile.rs

```diff
@@ -74,6 +76,23 @@ pub enum CompileMode {
     Doc { deps: bool },
 }
 
+#[derive(Clone, Copy, PartialEq, Eq)]
+pub enum MessageFormat {
+    Human,
+    Json
+}
+
+impl MessageFormat {
```

Comment:
Yep, I'll do this when I have time (soonish). I think I'll leave `json` instead of `json-v1`. `v1` was intended for "let's output compiler messages as is, and wrap them later", but we are already do the streaming.